core/src/main/java/greencity/filters/MySpecification.java

```diff
@@ -0,0 +1,60 @@
+package greencity.filters;
+
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import org.springframework.context.annotation.Scope;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+@Scope("prototype")
+public interface MySpecification<T> extends Specification<T> {
```

Comment:
Why interface? Why is this a @Component?

What I was explaining is, for example:
- you have `RatingStatisticsRepo extends JpaSpecificationExecutor<RatingStatistics>` -> allows you to call `findAll(Specification)`
- `RatingStatisticsRepo` is an interface -> you can't add logic
- but starting from Java 8 you can add `default` methods -> put the method that converts DTO to `Specification` in the interface/repo that uses this Specification